diff -ru ../old/gnome-sharp-2.16.0/gconf/GConf/gconf-sharp-2.0.pc.in gnome-sharp-2.16.0/gconf/GConf/gconf-sharp-2.0.pc.in
--- ../old/gnome-sharp-2.16.0/gconf/GConf/gconf-sharp-2.0.pc.in	2006-08-21 10:37:30.000000000 -0600
+++ gnome-sharp-2.16.0/gconf/GConf/gconf-sharp-2.0.pc.in	2007-12-07 11:50:41.000000000 -0700
@@ -7,4 +7,4 @@
 Description: GConf# - GConf .NET Binding
 Version: @VERSION@
 Requires: gnome-sharp-2.0
-Libs: -r:${libdir}/mono/@PACKAGE_VERSION@/gconf-sharp.dll -r:${libdir}/mono/@PACKAGE_VERSION@/gconf-sharp-peditors.dll
+Libs: -r:${libdir}/mono/@PACKAGE_VERSION@/gconf-sharp.dll
diff -ru ../old/gnome-sharp-2.16.0/gconf/GConf/Makefile.in gnome-sharp-2.16.0/gconf/GConf/Makefile.in
--- ../old/gnome-sharp-2.16.0/gconf/GConf/Makefile.in	2006-08-21 10:46:22.000000000 -0600
+++ gnome-sharp-2.16.0/gconf/GConf/Makefile.in	2007-12-07 11:55:17.000000000 -0700
@@ -231,10 +231,9 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-@ENABLE_GNOME_FALSE@TARGET = 
-@ENABLE_GNOME_TRUE@TARGET = $(ASSEMBLY) $(ASSEMBLY).config
-@ENABLE_GNOME_TRUE@pkgconfigdir = $(libdir)/pkgconfig
-@ENABLE_GNOME_TRUE@pkgconfig_DATA = gconf-sharp-2.0.pc
+TARGET = $(ASSEMBLY) $(ASSEMBLY).config
+pkgconfigdir = $(libdir)/pkgconfig
+pkgconfig_DATA = gconf-sharp-2.0.pc
 noinst_DATA = $(TARGET) $(POLICY_ASSEMBLIES)
 CLEANFILES = $(ASSEMBLY) $(ASSEMBLY).mdb gtk-sharp.snk $(POLICY_ASSEMBLIES) $(POLICY_CONFIGS)
 DISTCLEANFILES = gconf-sharp-2.0.pc $(ASSEMBLY).config
@@ -260,7 +259,7 @@
 	NotifyWrapper.cs 	\
 	Value.cs
 
-build_sources = $(addprefix $(srcdir)/, $(sources)) ../../AssemblyInfo.cs
+build_sources = $(addprefix $(srcdir)/, $(sources))
 @PLATFORM_WIN32_FALSE@GAPI_CDECL_INSERT = 
 @PLATFORM_WIN32_TRUE@GAPI_CDECL_INSERT = $(top_srcdir)/gapi-cdecl-insert --keyfile=gtk-sharp.snk $(ASSEMBLY)
 all: all-am
@@ -455,9 +454,12 @@
 gtk-sharp.snk: $(top_srcdir)/gtk-sharp.snk
 	cp $(top_srcdir)/gtk-sharp.snk .
 
-$(ASSEMBLY): $(build_sources) gtk-sharp.snk
+AssemblyInfo.cs: $(top_builddir)/AssemblyInfo.cs
+	cp $(top_builddir)/AssemblyInfo.cs .
+
+$(ASSEMBLY): $(build_sources) gtk-sharp.snk AssemblyInfo.cs
 	@rm -f $(ASSEMBLY).mdb
-	$(CSC) $(CSFLAGS) /out:$(ASSEMBLY) /target:library $(references) $(build_sources)
+	$(CSC) $(CSFLAGS) /out:$(ASSEMBLY) /target:library $(references) $(build_sources) AssemblyInfo.cs
 	$(GAPI_CDECL_INSERT)
 
 $(POLICY_ASSEMBLIES): $(top_builddir)/policy.config gtk-sharp.snk
