diff -Naur gnome-sharp-2.24.1/Makefile.in gnome-sharp-2.24.1-patched/Makefile.in
--- gnome-sharp-2.24.1/Makefile.in	2009-03-03 09:09:52.000000000 -0700
+++ gnome-sharp-2.24.1-patched/Makefile.in	2009-09-30 19:16:47.000000000 -0600
@@ -227,7 +227,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = sources art gnomevfs gnome gconf sample 
+SUBDIRS = sources art gnomevfs gconf sample 
 EXTRA_DIST = \
 	gtk-sharp.snk		\
 	gapi-cdecl-insert	\
