Only in .: build
diff -ru ../boo.old/default.build ./default.build
--- ../boo.old/default.build	2007-06-12 08:29:16.000000000 -0600
+++ ./default.build	2007-07-06 11:19:06.000000000 -0600
@@ -58,7 +58,7 @@
 	<property name="csc.define" value="NET_2_0"
 			if="${2 == version::get-major(framework::get-version(framework::get-target-framework()))}" />
 	
-	<target name="all" depends="booc, booi, booish, boo-pkgconfig, Boo.Lang.CodeDom, Boo.Lang.Useful, boo, Boo.Microsoft.Build.Tasks, update-vs2005-env">	
+	<target name="all" depends="booc, booi, booish, boo-pkgconfig, Boo.Lang.CodeDom, Boo.Lang.Useful, boo, Boo.Microsoft.Build.Tasks">	
 	</target>
 	
 	<target name="set-release-configuration">
