--- mono-1.9.1/configure.in
+++ mono-1.9.1/configure.in
@@ -44,8 +44,6 @@ libmono_cflags=""
 libmono_ldflags=""
 AC_SUBST(libmono_cflags)
 AC_SUBST(libmono_ldflags)
-export_ldflags=`(./libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
-AC_SUBST(export_ldflags)
 
 # Variable to have relocatable .pc files (lib, or lib64)
 reloc_libdir=`basename ${libdir}`
@@ -291,6 +289,10 @@ AC_LIBTOOL_WIN32_DLL
 #AC_DISABLE_FAST_INSTALL
 AM_PROG_LIBTOOL
 
+m4_ifdef([LT_OUTPUT], [LT_OUTPUT])
+export_ldflags=`(./libtool --config; echo eval echo \\$export_dynamic_flag_spec) | sh`
+AC_SUBST(export_ldflags)
+
 # Test whenever ld supports -version-script
 AC_PROG_LD
 AC_PROG_LD_GNU
