diff -ur ./frontend/libedit/term.c ../new/mono-debugger-0.31/frontend/libedit/term.c
--- ./frontend/libedit/term.c	2006-11-14 21:02:36.000000000 +0100
+++ ../new/mono-debugger-0.31/frontend/libedit/term.c	2006-11-14 21:59:09.000000000 +0100
@@ -947,7 +947,7 @@
 		Val(T_li) = tgetnum("li");
 		for (t = tstr; t->name != NULL; t++) {
 			/* XXX: some systems' tgetstr needs non const */
-			term_alloc(el, t, tgetstr(strchr(t->name, *t->name),
+			term_alloc(el, t, (char *)tgetstr(strchr(t->name, *t->name),
 			    &area));
 		}
 	}
@@ -1471,7 +1471,7 @@
 		}
 	if (t->name == NULL) {
 		/* XXX: some systems' tgetstr needs non const */
-		scap = tgetstr(strchr(*argv, **argv), &area);
+		scap = (char *)tgetstr(strchr(*argv, **argv), &area);
 	}
 	if (!scap || scap[0] == '\0') {
 		if (!silent)
