--- libgdiplus-1.1.13.2/src/graphics.c	Tue Jan 24 18:32:11 2006
+++ libgdiplus-1.1.13.2/src/graphics.c	Wed Jan 18 14:40:42 2006
@@ -468,7 +468,7 @@
 	
 	
 	*graphics = gdip_graphics_new(surface);
-	graphics->dpi_x = *graphics->dpi_y = gdip_get_display_dpi ();
+	(*graphics)->dpi_x = (*graphics)->dpi_y = gdip_get_display_dpi ();
 	cairo_surface_destroy (surface);
 	
 	(*graphics)->bounds.Width = width;
