Index: BooShell/Makefile
===================================================================
--- BooShell/Makefile	(revision 91600)
+++ BooShell/Makefile	(working copy)
@@ -80,7 +80,7 @@
 	make pre-install-local-hook prefix=$(prefix)
 	mkdir -p $(INSTALL_DIR)
 	cp $(ASSEMBLY) $(ASSEMBLY_MDB) $(INSTALL_DIR)
-	mkdir -p $(prefix)/lib/pkgconfig
+	mkdir -p $(DESTDIR)$(prefix)/lib/pkgconfig
 	make post-install-local-hook prefix=$(prefix)
 
 uninstall-local: $(ASSEMBLY) $(ASSEMBLY_MDB)
